File: Children-Competition-C#/Service/EventService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Concurs.Domain;
using Concurs.Repository;

namespace Concurs.Service
{
    internal class EventService
    {
        private EventDBRepository eventDBRepository;
        public EventService(EventDBRepository eventDBRepository)
        {
            this.eventDBRepository = eventDBRepository;
        }

        public List<Event> FindAll()
        {
            return (List<Event>)this.eventDBRepository.FindAll();
        }

        public Event FinyByNameAge(string Name, int AgeInRange)
        {
            return this.eventDBRepository.FindByNameAge(Name, AgeInRange);
        }

        public int AddEnrolledToEvent(Guid id)
        {
            return this.eventDBRepository.AddEnrolledToEvent(id);
        }
    }
}

File: Children-Competition-Java/mpp-proiect-java-ABNegrea/Problema5/src/main/java/Concurs/Service/EventService.java
package Concurs.Service;

import Concurs.Domain.Event;
import Concurs.Repository.EventDBRepository;

import java.util.List;
import java.util.UUID;

public class EventService {
    private EventDBRepository eventRepo;

    public EventService(EventDBRepository eventRepo) {
        this.eventRepo = eventRepo;
    }

    public List<Event> findAll()
    {
        return eventRepo.findAll().stream().toList();
    }

    public Event findByNameAge(String name, int ageInRange){
        return eventRepo.findByNameAge(name,ageInRange);
    }

    public int addEnrolledToEvent(UUID id){
        return this.eventRepo.addEnrolledToEvent(id);
    }
}
